torchao/prototype/intx/Intx.py

```diff
@@ -0,0 +1,255 @@
+import functools
+import math
+from collections import defaultdict
+from typing import Any, Callable, Dict, Optional, Tuple, Union, List
+
+import torch
+from torch._dynamo.comptime import comptime
+
+from torch.utils._python_dispatch import return_and_correct_aliasing
+from .bitpacking import pack, unpack, numbits
+from torchao.dtypes.utils import (
+    _implements,
+    _ATEN_OP_OR_TORCH_FN_TABLE,
+    _register_layout_cls,
+    _get_layout_tensor_constructor,
+)
+
+
+aten = torch.ops.aten
+def implements(aten_ops_or_torch_fn):
```

Comment:
this can be updated with https://github.com/pytorch/ao/blob/de4a1fb3b1f71e2f61b84dfdc96e7d704ff72208/torchao/dtypes/affine_quantized_tensor.py#L290 as well